build_web_compilers/lib/src/dev_compiler_bootstrap.dart

```diff
@@ -317,8 +317,8 @@ var baseUrl = (function () {
   // base href should start with "/"
   if (typeof document !== 'undefined') {
     var el = document.getElementsByTagName('base');
-    if (el && el[0] && el[0].href && el[0].href.startsWith('/')){
-    	return el[0].href;
+    if (el && el[0] && el[0].getAttribute("href") && el[0].getAttribute("href").startsWith("/")){
+      return el[0].href;
```

Comment:
should this also be `getAttribute("href")`?

Reply:
I removed unnecessary check for `el[0].href`.

Reply:
maybe just create a local variable containing the href since its accessed 3 times?

Reply:
I'm referring to how it's inconsistent. We have two place uses `.getAttribute("href")` and one using `.href`. Why are we picking one or the other in each place?

If the `.getAttribute("href")` pattern is for when we aren't sure if it exists, and `.href` pattern is for when we know it exists, then we should use `.href.startsWith("/")` above since in that location we know it exists...

If there is some other condition making us choose one pattern or the other - what is that condition?

Reply:
@natebosch 
`el[0].href` returns `http://example.com/abc/` , so we can't perform a successful `startsWith("/")` check on `el[0].href`, whereas `el[0].getAttribute("href")` will return `/abc/` and `el[0].getAttribute("href").startsWith("/")` will return `true`.

Reply:
@jakemac53 
`el[0].getAttribute("href")` is used twice while checking condition
`el[0].href` is used only once while returning value.

Here I'm trying to make sure it works and doesn't fail.

Reply:
Thanks for the explanation!